File: Tellma/ClientApp/src/app/data/dto/get-response.ts
// tslint:disable:variable-name
import { EntityWithKey } from '../entities/base/entity-with-key';
import { Entity } from '../entities/base/entity';
import { EntitiesResponse } from './entities-response';

export interface GetResponse<TEntity extends Entity = EntityWithKey> extends EntitiesResponse<TEntity> {
  Skip: number;
  Top: number;
  OrderBy: string;
  Desc: boolean;
  TotalCount: number;
}

File: Tellma.Api.Web.Dto/GetResponse.cs
using Tellma.Model.Common;

namespace Tellma.Api.Dto
{
    /// <summary>
    /// Represents a response of raw entities for save + some search metadata.
    /// </summary>
    public class GetResponse<TEntity> : EntitiesResponse<TEntity> where TEntity : Entity
    {
        public int Skip { get; set; }

        public int Top { get; set; }

        public string OrderBy { get; set; }

        public int? TotalCount { get; set; }
    }
}
